src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -181,6 +181,19 @@ private void MethodChecks(MethodDeclarationSyntax syntax, Binder withTypeParamsB
                 }
             }
 
+            // warn for CancellationToken parameters in async-iterators with no parameter decorated with [EnumeratorCancellation]
+            var location = this.Locations[0];
+            var cancellationTokenType = DeclaringCompilation.GetWellKnownType(WellKnownType.System_Threading_CancellationToken);
+            var iAsyncEnumerableType = DeclaringCompilation.GetWellKnownType(WellKnownType.System_Collections_Generic_IAsyncEnumerable_T);
+            if (IsAsync &&
+                ReturnType.OriginalDefinition.Equals(iAsyncEnumerableType) &&
+                (Bodies.blockBody != null || Bodies.arrowBody != null) &&
+                ParameterTypesWithAnnotations.Any(p => p.Type.Equals(cancellationTokenType)) &&
+                !Parameters.Any(p => p.HasEnumeratorCancellationAttribute))
```

Comment:
What if `[EnumeratorCancellation]` is applied to the wrong parameter?

Reply:
Then we produce no warning. There is already a warning if you place an `[EnumeratorCancellation]` attribute on a parameter where it can't have an effect (for example, a parameter of some other type). I don't think it helps to produce two warnings in such case, but I don't feel strongly about it.

Reply:
No need for a second warning. Consider adding a comment here why we don't need to check which parameter has the attribute.

---
In reply to: [278667755](https://github.com/dotnet/roslyn/pull/35254#discussion_r278667755) [](ancestors = 278667755)